### DISCUSSION TITLE
Adding a root url path for app with assets

### DISCUSSION THREAD

**Discussion by vanishingESCkey** [mojolicious/mojo#2127] [Q&A] [answered]
I have a Mojolicious app and need to "move" all routes to work with root path "/foo".  Already found a `partial` attribute for routes that can help me:

```
$r = $r->any( '/foo')->partial( 1 );
# then use $r for existing routes
$r->get("/pages/home")->to("pages#home");
```

The problem is that I use "Mojolicious::Plugin::AssetPack" for serving css  & js files and can't figure out a smart way how to add same "/foo" prefix for assets urls. Maybe there is some better solution than `partial` which can add such url prefix for all routes defined in my app?

**Comment by kraih** [comment, marked as answer]
I don't think `partial` contributes anything of value there. For `Mojolicious::Plugin::AssetPack` there should be a `route` option somewhere that can be used to set a prefix. And you forgot about static files, in newer versions of Mojolicious those can also have aprefix configured with https://metacpan.org/pod/Mojolicious::Static#prefix.

**Reply by vanishingESCkey** [reply]
Thanks, I completely missed that `route` option, very useful.

The solution with `partial` was copy pasted from stackoverflow, but leaving just `any` without `partial` seems to do the same job.
